test/functional/apps/dashboard/_dashboard_queries.js

```diff
@@ -0,0 +1,97 @@
+import expect from 'expect.js';
+
+import { PIE_CHART_VIS_NAME } from '../../page_objects/dashboard_page';
+
+/**
+ * Test the querying capabilities of dashboard, and make sure visualizations show the expected results, especially
+ * with nested queries and filters on the visualizations themselves.
+ */
+export default function ({ getService, getPageObjects }) {
+  const retry = getService('retry');
+  const dashboardVisualizations = getService('dashboardVisualizations');
+  const PageObjects = getPageObjects(['dashboard', 'header', 'visualize']);
+
+  describe('dashboard queries', function describeIndexTests() {
+    before(async function () {
+      return PageObjects.dashboard.initTests();
+    });
+
+    after(async function () {
+      // avoids any 'Object with id x not found' errors when switching tests.
+      await PageObjects.header.clickVisualize();
+      await PageObjects.visualize.gotoLandingPage();
+      await PageObjects.header.clickDashboard();
+      await PageObjects.dashboard.gotoDashboardLandingPage();
+    });
+
+    it('Nested visualization query filters data as expected', async () => {
+      // This flip between apps fixes the url so state is preserved when switching apps in test mode.
+      // Without this flip the url in test mode looks something like
+      // "http://localhost:5620/app/kibana?_t=1486069030837#/dashboard?_g=...."
+      // after the initial flip, the url will look like this: "http://localhost:5620/app/kibana#/dashboard?_g=...."
+      await PageObjects.header.clickVisualize();
```

Comment:
If I understand correctly, this must be done before any Dashboard test is run, should we consider moving it to `before` so if we reordered these tests it wouldn't break?

Reply:
Yeaaa, I honestly don't know why, but moving these to the `before` thing fails to remove the t=123 part of the URL.  I wonder if there is some other `before` call that is doing the adding of it.  Agree it's ugly and should really be in the before (or just part of initTests since all dashboard tests essentially do this).